app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CreateEventActivity.kt

```diff
@@ -242,17 +291,67 @@ class CreateEventActivity : ActivityForAccessingTumOnline<CreateEventResponse>(R
         }
         event.description = description
         this.event = event
+        events = generateEvents()
+
+        for (curEvent in events!!) {
+            val apiCall = apiClient.createEvent(curEvent, null)
+            fetch(apiCall)
+        }
+    }
 
-        val apiCall = apiClient.createEvent(event, null)
-        fetch(apiCall)
+    /**
+     * generates a list of events that are added to the calendar
+     * depending on the repeat-setting
+     */
+    private fun generateEvents(): List<CalendarItem> {
+        if (!repeats) {
+            return listOf(this.event!!)
+        }
+        val items = ArrayList<CalendarItem>()
+        seriesId = generateSeriesId()
+
+        // event ends after n times
+        if (endAfterRadioBtn.isChecked) {
+            val numberOfEvents = eventRepeatsTimes.text.toString().toInt()
```

Comment:
You need a sanity check here, what if its `0`?

btw. you have the same code in `createEventButton.setOnClickListener`. I'd suggest only fetching the value once and then adding it as a function parameter to  `createEvent` - this makes debugging easier.

Reply:
Also: if you do it as a parameter, then the error checking becomes easier and your control flow in the onclicklisteneer can be reduced as well.

Reply:
The sanity check takes place before this function is called (in `createEventButton.setOnClickListener`) this is properly not best practice, but taking the number as a parameter would also be quite annoying, since there would need to be parameters for when the event repeats n times and for when it ends after a specific date. I could however overload the method or use class attributes. Which option would you prefer?

Reply:
Lets keep it as is and refine later. I think doing this with an overloaded method is the best way in java.